homeassistant/components/mqtt/vacuum/schema_legacy.py

```diff
@@ -171,25 +190,40 @@ class MqttVacuum(MqttEntity, VacuumEntity):
     _entity_id_format = ENTITY_ID_FORMAT
     _attributes_extra_blocked = MQTT_LEGACY_VACUUM_ATTRIBUTES_BLOCKED
 
-    def __init__(self, hass, config, config_entry, discovery_data):
+    _encoding: str | None
+    _fan_speed_list: list[str]
+    _qos: bool
+    _retain: bool
+    _payloads: dict[str, str]
+    _send_command_topic: str | None
+    _set_fan_speed_topic: str | None
+    _state_topics: dict[str, Any]
+    _templates: dict[str, Callable[..., ReceivePayloadType]]
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        config: ConfigType,
+        config_entry: ConfigEntry,
+        discovery_data: DiscoveryInfoType | None,
+    ) -> None:
         """Initialize the vacuum."""
-        self._cleaning = False
-        self._charging = False
-        self._docked = False
-        self._error = None
-        self._status = "Unknown"
-        self._battery_level = 0
-        self._fan_speed = "unknown"
-        self._fan_speed_list = []
+        self._cleaning: bool = False
+        self._charging: bool = False
+        self._docked: bool = False
+        self._error: str | None = None
+        self._status: str = "Unknown"
+        self._battery_level: int = 0
+        self._fan_speed: str | None = "unknown"
 
         MqttEntity.__init__(self, hass, config, config_entry, discovery_data)
 
     @staticmethod
-    def config_schema():
+    def config_schema() -> vol.Schema:
```

Comment:
Would it be more useful to annotate `DISCOVERY_SCHEMA_LEGACY`? That way that type information would propagate everywhere necessary..?

Reply:
The type matches